[BUG] Fix #145 drawdown overlaps

jlopezpena opened the pull request:
Fix the drawdown overlaps described in #145, also making the `get_drawdown` function more efficient.

Add a regression test that checks that the recovery of each drawdown occurs before the peak of the next (when ordered by date). 

Note: The test requires being online to pull the FB stock data (or have it cached). I couldn't find any guidelines about tests. If that is not appropriate let me know and I will work on getting an offline version of it.

twiecki commented:
Awesome, thanks for pulling that together. The question re FB is a good one. Can you change it to SPY which we provide a cached version of?

jlopezpena commented:
Changed the test to use SPY, and lightly modified the check in order to get more informative messages in case of error (a test error should now show the dates for the overlapping drawdown periods rather than the uninformative `True does not equal False` that was showing before).

Reproducing the error for SPY required inspecting over a larger number of drawdowns (20 rather than 10)

Patch is also rebased on top of the latest master branch

twiecki commented:
LGTM, thanks!
